Vagrantfile

```diff
@@ -21,192 +21,277 @@
 # specific language governing permissions and limitations
 # under the License.
 
+define_opts = {
+  autostart: false
+}.freeze
+
 Vagrant.configure(2) do |config|
-  config.vm.define "ubuntu-1404" do |config|
-    config.vm.box = "elastic/ubuntu-14.04-x86_64"
-    ubuntu_common config
+
+  config.vm.provider 'virtualbox' do |vbox|
+    # Give the box more memory and cpu because our tests are beasts!
+    vbox.memory = Integer(ENV['VAGRANT_MEMORY'] || 8192)
+    vbox.cpus = Integer(ENV['VAGRANT_CPUS'] || 4)
   end
-  config.vm.define "ubuntu-1604" do |config|
-    config.vm.box = "elastic/ubuntu-16.04-x86_64"
-    ubuntu_common config, extra: <<-SHELL
-      # Install Jayatana so we can work around it being present.
-      [ -f /usr/share/java/jayatanaag.jar ] || install jayatana
-    SHELL
+
+  # Vagrant and ruby stdlib expand '.' differently, so look for the build
+  # in the directory containing the Vagrantfile
+  vagrantfile_dir = File.expand_path('..', __FILE__)
+
+  # Switch the default share for the project root from /vagrant to
+  # /elasticsearch because /vagrant is confusing when there is a project inside
+  # the elasticsearch project called vagrant....
+  config.vm.synced_folder vagrantfile_dir, '/vagrant', disabled: true
+  config.vm.synced_folder vagrantfile_dir, '/elasticsearch'
+
+  # Expose project directory. Note that VAGRANT_CWD may not be the same as Dir.pwd
+  PROJECT_DIR = ENV['VAGRANT_PROJECT_DIR'] || Dir.pwd
```

Comment:
What is the "project directory" compared to the vagrantfile_dir above? I don't think we need to (or should) expose anything but the dir containing the tests, especially since the fallback is the working dir, for which nothing in tests can rely on.

Reply:
This project directory [gets set by gradle](https://github.com/elastic/elasticsearch/blob/2c24ac742634e8c0af849f74d885e0efd52f3c0f/buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy#L283) as the directory of the gradle project that applies the plugin and runs vagrant shell commands, so in this case `qa/vagrant`. This is where the bats tests run because everything they need is self-contained in the project. The new packaging tests need the entire gradle project. Once the bats tests have all been ported I think we can get rid of the project directory set here.

Reply:
But what good is pwd here as a fallback? Seems it would just be horribly broken assuming the working dir contains the tests, and we should instead fail if the env is not set.

Reply:
And also if that is the case this seems unrelated to the windows work? ie adding validation/error on it missing could be done in a separate PR.

Reply:
Right, it's unrelated to the windows stuff, it's just [relocated from below](https://github.com/elastic/elasticsearch/pull/27682/files#diff-23b6f443c01ea2efcb4f36eedfea9089L85). I'm not sure what the rationale is for falling back to pwd - I'd guess it's so that if you're making vagrant commands outside of gradle, you can just run them from the project directory and it'll mount it without you having to set the env var.

It would definitely be good to validate it, but it's going to be removed once all the bats tests are ported so I think it's okay to leave as-is. Happy to add some validation though if we think it's important in the meantime

Reply:
>  it's just relocated from below

Aha! This is why minimal changes are important. The PR made it look new, thus my comments.